packages/next/server/node-web-streams-helper.ts

```diff
@@ -181,6 +181,11 @@ function createHeadInsertionTransformStream(
           freezing = true
           inserted = true
         }
+        if (process.env.NODE_ENV !== 'production' && insertion && !inserted) {
+          console.error(
+            `server inserted HTML couldn't be inserted into the stream. You are missing '<head/>' element in your layout - please add it.`
+          )
+        }
```

Comment:
Alternatively, we could also inject it to the end of the stream and ignore the missing `<head/>`. But I've figured out that most commonly, people would like for this stuff to be inserted into `<head/>` so it might be better to "fail fast" and just provide a warning here